Packs/CortexAttackSurfaceManagement/Scripts/InferWhetherServiceIsDev/InferWhetherServiceIsDev_test.py

```diff
@@ -9,47 +9,112 @@ def test_canonicalize():
+                          ])
+def test_get_indicators_from_key_value_pairs(tags_raw, matches):
+    from InferWhetherServiceIsDev import get_indicators_from_key_value_pairs
+    from InferWhetherServiceIsDev import is_dev_indicator
 
-    # dev indicator with varying keys
-    assert is_dev_according_to_key_value_pairs([{"Key": "env", "Value": "dev"},
-                                                {"Key": "Name", "Value": "rdp_server"},
-                                                {"Key": "test", "Value": ""}])
-    assert is_dev_according_to_key_value_pairs([{"Key": "ENVIRONMENT", "Value": "TEST"}])
+    assert get_indicators_from_key_value_pairs(tags_raw, is_dev_indicator) == matches
 
-    # pre-prod counts as dev and not as prod
-    assert is_dev_according_to_key_value_pairs([{"Key": "Stage", "Value": "status - preprod"}])
 
-    # no dev indicator
-    assert not is_dev_according_to_key_value_pairs([{"Key": "env", "Value": "prod"}])
-    assert not is_dev_according_to_key_value_pairs([{"Key": "dev", "Value": "my name"}])
+def test_is_dev_indicator():
+    from InferWhetherServiceIsDev import is_dev_indicator
 
-    # conflicting indicators
-    assert not is_dev_according_to_key_value_pairs([{"Key": "env", "Value": "prod"},
-                                                    {"Key": "env", "Value": "dev"}])
+    # Test Dev Matches
+    assert is_dev_indicator('dev')
+    assert is_dev_indicator('uat')
+    assert is_dev_indicator('non-prod')
+    assert is_dev_indicator('noprod')
 
-    # extra arguments ok
-    assert is_dev_according_to_key_value_pairs([{"Key": "ENVIRONMENT", "Source": "AWS", "Value": "TEST"}])
+    # Test no match
+    assert not is_dev_indicator('devops')
+    assert not is_dev_indicator('prod')
+    assert not is_dev_indicator('pr')
 
 
-def test_is_dev_according_to_classifications():
-    from InferWhetherServiceIsDev import is_dev_according_to_classifications
+def test_is_prod_indicator():
+    from InferWhetherServiceIsDev import is_prod_indicator
 
-    assert is_dev_according_to_classifications(["SshServer", "DevelopmentEnvironment"])
-    assert not is_dev_according_to_classifications(["RdpServer", "SelfSignedCertificate"])
+    # Test Dev Matches
+    assert is_prod_indicator('pr')
+    assert is_prod_indicator('prod')
+
+    # Test no Matches
+    assert not is_prod_indicator('non-prod')
+    assert not is_prod_indicator('staging')
+
+
+@pytest.mark.parametrize('classifications, matches', [(["SshServer", "DevelopmentEnvironment"],
+                                                       ["DevelopmentEnvironment"]),
+                                                      (["SshServer"], [])])
+def test_get_indicators_from_external_classification(classifications, matches):
+    from InferWhetherServiceIsDev import get_indicators_from_external_classification
+
+    assert get_indicators_from_external_classification(classifications) == matches
+
+
+@pytest.mark.parametrize('external, internal, reason',
```

Comment:
Additional comment, @ptoman-pa I think this is harder to read on the PR but I feel your sentiment, I would use a dictionary here if that helps, I would advice against extrapolating out this many test for just different parameters.

```suggestion
reason_data = {
  "external": {
    "none": [],
    "non_prod": ["DevelopmentEnvironment"],
    "prod":["ProductionEnvironment"]
    },
  "internal": {
    "non_prod": [{"Key": "env", "Value": "non-prod", "Source": "AWS"}],
    "prod": [{"Key": "env", "Value": "prod", "Source": "AWS"}],
    "unknown_env_case": [{"Key": "env", "Value": "custom-tag", "Source": "AWS"}],
    "multiple_prod":[{"Key": "env", "Value": "non-prod", "Source": "AWS"}, {"Key": "stage", "Value": "sbx", "Source": "GCP"}]
  }
}

@pytest.mark.parametrize('external, internal', 'reason'
                         [(reason_data["external"]["non_prod"], reason_data["internal"]["non_prod"], "match on external classification of DevelopmentEnvironment and tag {env: non-prod} from AWS"))])
```